tests/generic-java-signatures/i15385/Lib.scala

```diff
@@ -0,0 +1,24 @@
+class Loc(val idx: Int) extends AnyVal
+
+class Foo:
+  def testNoParam[A <: Int]: A = 1.asInstanceOf[A]
+  def testSingleParam[A <: Int](a: A): A = 2.asInstanceOf[A]        // <A:Ljava/lang/Object;>(I)I
```

Comment:
Shouldn't we drop the generic signature here entirely? In the end it's just int => int

Reply:
I think it's a matter of taste a this point.  From Java, there's no more link between any passed type argument and the term argument, but maybe there is still a type argument you can provide - so that's why I kept it, as well as it being easier to implement it this way.  Unless we have a real reason to drop it, I think this is good.